Guide/src/reference/architecture/openhcl_servicing.md

```diff
@@ -0,0 +1,258 @@
+
+### Without Keepalive
+
+When keepalive is not enabled or not available:
+
+1. NVMe devices are cleanly shut down
+2. VFIO device handles are closed (triggering FLR - Function Level Reset)
+3. All device state is lost
+4. On restore, devices must be fully reinitialized
+5. Guest OS must handle device reappearance and potential I/O errors
+
+## Compatibility and Versioning
+
+### Saved State Format
+
+The servicing state uses Protocol Buffers (`Protobuf`) for serialization:
+- Forward and backward compatibility with field numbering
+- Schema evolution support
+- Cross-version restore capability
+
+### Compatibility Handling
+
+OpenHCL includes logic to handle state from different versions:
+
+#### `fix_pre_save()`
+- Updates state before saving to ensure compatibility with older versions
+- Handles legacy field conversions
+- Maintains VMBus relay compatibility with release branches
+
+#### `fix_post_restore()`
+- Updates state after loading from older versions
+- Converts legacy formats to current format
+- Ensures proper field population for current code
+
+## Configuration
+
+Servicing behavior is controlled by several configuration options:
+
+### NVMe Keepalive Configuration
+
+- `host,privatepool`: Enable keepalive if both host and private pool support it
+- `nohost,privatepool`: Private pool available but host keepalive disabled
+- `nohost,noprivatepool`: Keepalive fully disabled
+
+### Test Scenarios
+
+For testing servicing behavior:
+- `SaveStuck`: Causes save operation to wait indefinitely
+- `SaveFail`: Forces save operation to fail
+- These help test timeout handling and failure recovery
+
+## Error Handling
+
+Servicing operations include robust error handling:
+
+### Save Failures
+- State units that fail to save are logged
+- Overall servicing operation may still succeed if critical state is saved
+- Non-critical failures are logged but don't block servicing
+
+### Restore Failures
+- Critical component failures prevent VM startup
+- Error messages indicate which component failed
+- Detailed logging helps diagnose issues
+
+### Timeout Handling
+- Host provides a deadline for save operation
+- OpenHCL attempts to complete save before deadline
+- If deadline is exceeded, host may force termination
+
+## Servicing State Components
+
+### State Units
+
+State units represent saveable/restorable VM components:
+- Each has a unique name identifier
+- Saved state is stored as a blob
+- Units can be saved/restored independently
+
+### Correlation ID
+
+The correlation ID:
+- Tracks the servicing operation end-to-end
+- Used in all logging and tracing during servicing
+- Helps correlate events across host and guest
+- Persists across the servicing operation
+
+## Implementation Details
+
+### Mesh RPC Architecture
+
+Servicing uses mesh RPC for coordinating async operations:
+- Request/response pattern for save operations
+- Graceful shutdown coordination
+- Timeout and cancellation support
```

Comment:
This isn't super relevant. Rather, create a new page that describes Mesh and the types of things it can be used for.